src/RedisSessionHandler.php

```diff
@@ -137,9 +137,7 @@ public function create_sid()
     public function read($session_id)
     {
         if ($this->mustRegenerate($session_id)) {
-            session_id($session_id = $this->create_sid());
-
-            setcookie($this->cookieName, $session_id);
+            session_regenerate_id(true);
```

Comment:
I used to call `session_regenerate_id` in the beginning, but it won't work on 7.1, see https://github.com/1ma/RedisSessionHandler/issues/3#issuecomment-281320830

Please revert to your original proposal :P